pkg/client/unversioned/helper.go

```diff
@@ -257,24 +257,31 @@ func NewOrDie(c *Config) *Client {
 // running inside a pod running on kuberenetes. It will return an error if
 // called from a process not running in a kubernetes environment.
 func InClusterConfig() (*Config, error) {
-	token, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/" + api.ServiceAccountTokenKey)
-	if err != nil {
-		return nil, err
-	}
-	tlsClientConfig := TLSClientConfig{}
-	rootCAFile := "/var/run/secrets/kubernetes.io/serviceaccount/" + api.ServiceAccountRootCAKey
-	if _, err := util.CertPoolFromFile(rootCAFile); err != nil {
-		glog.Errorf("Expected to load root CA config from %s, but got err: %v", rootCAFile, err)
+	config := &Config{}
+	if _, err := os.Stat("/var/run/secrets/kubernetes.io/serviceaccount"); err != nil {
```

Comment:
is that a good method in general for figuring out whether to use TLS to connect to a service from inside a pod? I didn't know if that correlation was dependable or not... I don't think it works for user-created services, does it?